### PR TITLE
📦️(frontend) integrate changesets to publish npm packages

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We need to publish frontend components we use for Warren UI so that plugins developer can use them in their own plugins.

## Proposal

As recommended by TurboJS, we've integrated the changesets tool to handle frontend related changes and packages publication to npm.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -76,6 +76,7 @@ services:
       - dev
     volumes:
       - ./src/frontend:/app
+      - .git:/app/.git
```
Comment (lebaudantoine) [inline docker-compose.yml:79]: I am not totally sure to get this trick 🧐.

File: src/frontend/Dockerfile [hunk 2]
```diff
@@ -34,6 +34,11 @@ WORKDIR /app
 # -- Development --
 FROM core as development
 
+# Install Git (required for changeset)
+RUN apt-get update && \
+    apt-get install -y git && \
+    rm -rf /var/lib/apt/lists/*
+
```
Comment (lebaudantoine) [inline src/frontend/Dockerfile:41]: Let's discuss it, to understand this requirement!

**Review by lebaudantoine — APPROVED** [review]
LGTM

File: src/frontend/Dockerfile (hunk 2 above)
Comment (jmaupetit) [reply src/frontend/Dockerfile:41]: Git is required by changesets. If not installed the CLI fails while booting.

File: docker-compose.yml (hunk 1 above)
Comment (jmaupetit) [reply docker-compose.yml:79]: Git and a git history are required for the `changeset` CLI to run.

Comment (jmaupetit) [reply docker-compose.yml:79]: That's the best I was able to do.

**Pull request merged**